pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/segment/SegmentFlushThresholdComputer.java

```diff
@@ -157,13 +182,25 @@ public int computeThreshold(StreamConfig streamConfig, CommittingSegmentDescript
       }
     }
     targetSegmentNumRows = capNumRowsIfOverflow(targetSegmentNumRows);
+    targetSegmentNumRows = applySegmentSizeVariation(segmentSizeVariation, targetSegmentNumRows);
```

Comment:
This should happen before the cap, or it can overflow

Reply:
done